src/Service/ScheduledTask/OldNostoDataCleanupScheduledTaskHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Nosto\NostoIntegration\Service\ScheduledTask;
+
+use DateTime;
+use Nosto\NostoIntegration\Model\ConfigProvider;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Api\Context\SystemSource;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\AndFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\ContainsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\RangeFilter;
+use Shopware\Core\Framework\MessageQueue\ScheduledTask\ScheduledTaskHandler;
+use Throwable;
+
+class OldNostoDataCleanupScheduledTaskHandler extends ScheduledTaskHandler
+{
+    public function __construct(
+        EntityRepository $scheduledTaskRepository,
+        private readonly EntityRepository $mappingRepository,
+        private readonly ConfigProvider $configProvider,
+        private readonly LoggerInterface $logger,
+    ) {
+        parent::__construct($scheduledTaskRepository);
+    }
+
+    public static function getHandledMessages(): iterable
+    {
+        return [OldNostoDataCleanupScheduledTask::class];
+    }
+
+    public function run(): void
+    {
+        try {
+            $isNostoDataCleanupEnabled = $this->configProvider->isOldNostoDataCleanupEnabled();
+            $monthPeriod = $this->configProvider->getOldNostoDataCleanupPeriod();
+
+            if ($isNostoDataCleanupEnabled && $monthPeriod) {
+                $numberOfMonthsBeforeToday = new DateTime(' - ' . $monthPeriod . ' month');
+
+                // Here we have context-less process
+                $context = new Context(new SystemSource());
+                $criteria = new Criteria();
+                $criteria->addFilter(
+                    new AndFilter([
+                        new RangeFilter(
+                            'createdAt',
+                            [
+                                'lt' => $numberOfMonthsBeforeToday->format(Defaults::STORAGE_DATE_FORMAT),
+                            ],
+                        ),
+                        new ContainsFilter('mapping_table', 'cart'),
+                    ]),
+                );
+
+                $idSearchResult = $this->mappingRepository->searchIds($criteria, $context);
```

Comment:
im just wondering if you have a table like nordfishing thats like 30mill ids to be returned seems to me that its a lot of data in memory? can we just execute straight delete query?

Reply:
In my database, this table has 333 rows. As for deleting rows, the changes I made are implemented in the `OldJobCleanupScheduledTaskHandler` class.

Reply:
I understand that you may have 333 rows, but looking at nordfishing its 33million .... on staging :D so might be good to consider that. With Old jobs its different as that would never be a huge amount

Reply:
Okay, I agree with you that fetching a large number of rows and then deleting them can significantly slow down the process. I think it's better to delete directly without fetching.